### DISCUSSION TITLE
benchmark should use monotonic time

### DISCUSSION THREAD

**Discussion by codefromthecrypt** [jedisct1/libsodium#1253] [General]
I think this doesn't really need the discussion first, but just in case:

`now()` in benchmarks compiles to wasm that uses the real time clock, not the monotonic one.
https://github.com/jedisct1/libsodium/blob/6d566070b48efd2fa099bbe9822914455150aba9/test/default/cmptest.h#L119 

I think it should be something like `clock_gettime(CLOCK_MONOTONIC, &ts)` instead, to ensure better accuracy.

**Comment by jedisct1** [comment]
Done, thanks! But fo WASI, it doesn't make a difference.

**Reply by codefromthecrypt** [reply]
cheers yeah it was more the old telemetry person inside me than something that would have changed anything :D
